### PR TITLE
fix(ng-xform): unsubscribing from searchHandlers 

### PR DESCRIPTION
**Pull request by nobreconfrade**
Simple fix to unsubscribe from select-field `searchHandlers` subscriptions.

### CODE REVIEW

File: src/ng-xform/select-field/select-field.component.ts [hunk 1]
```diff
@@ -48,27 +52,32 @@ export class SelectFieldComponent extends BaseDynamicFieldComponent<SelectField>
...
-        if (value === undefined || value === null) {
-          this.updateOptionLabel();
+    this.subscriptions.push(
+      this.viewModel.subscribe((value: any) => {
+        if (this.field.searchByValueKeyHandler) {
+          if (value === undefined || value === null) {
+            this.updateOptionLabel();
+            return;
+          }
+          this.field.searchByValueKeyHandler(value).subscribe((val: any) => {
```
Comment (lvoliveira) [inline src/ng-xform/select-field/select-field.component.ts]: Make this subscription with a switchMap

**Review by lvoliveira — CHANGES_REQUESTED** [review]

File: src/ng-xform/select-field/select-field.component.ts (hunk 1 above)
Comment (lvoliveira) [reply src/ng-xform/select-field/select-field.component.ts]:
```suggestion
          this.viewModel.pipe(
      switchMap((value: any) => {
        this.select.writeValue(value);
        this.updateOptionLabel();
        if (this.field.searchByValueKeyHandler) {
          if (value === undefined || value === null) {
            this.updateOptionLabel();
            return of(null);;
          }
          return this.field.searchByValueKeyHandler(value);
        } else {
          return of(null);
        }
      }),
      tap((val: any) => {
        if (val && this.field.searchHandler) {
          const oldValue = this.optionValues;
          this.optionValues = [val];
          this.select.ngOnChanges({ items: new SimpleChange(oldValue, this.optionValues, !this.optionValues)});
        }
        // this.select.writeValue(val);
        // this.updateOptionLabel();
      })

    ).subscribe();
```

Comment (nobreconfrade) [reply src/ng-xform/select-field/select-field.component.ts]: changed

**Review by lvoliveira — APPROVED** [review]

**Review by igortg — APPROVED** [review]

File: src/ng-xform/select-field/select-field.component.ts [hunk 2]
```diff
@@ -48,27 +52,32 @@ export class SelectFieldComponent extends BaseDynamicFieldComponent<SelectField>
...
-    })
-  }
+        this.select.writeValue(value);
+        this.updateOptionLabel();
+      }));
+    }
+
+    ngOnDestroy(): void {
+      this.subscriptions.forEach(sub => sub.unsubscribe());
+    }
```
Comment (lincrosenbach) [inline src/ng-xform/select-field/select-field.component.ts:89]: Well done! :1st_place_medal:

**Review by lincrosenbach — APPROVED** [review]

**Review by igortg — APPROVED** [review]

**Pull request merged**